apps/SceneClustering/domset/domset.cc

```diff
@@ -0,0 +1,791 @@
+      pcCentre.pos += (p.pos / numPoints);
+    }
+  }
+  // calculation the normalization scale
+  const float avgDist = totalDist / numPoints;
+  std::cerr << "Total distance = " << totalDist << std::endl;
+  std::cerr << "Avg distance = " << avgDist << std::endl;
+  normScale = 1.f / avgDist;
+  std::cerr << "Normalization Scale = " << normScale << std::endl;
+
+// normalizing the distances on points
+#if DOMSET_USE_OPENMP
+#pragma omp parallel for
+#endif
+  for_parallel(i, numPoints)
+  {
+    points[i].pos = (points[i].pos - pcCentre.pos) * normScale;
+  }
+
+  // normalizing the camera center positions
+  const size_t numViews(views.size());
+#if DOMSET_USE_OPENMP
+#pragma omp parallel for
+#endif
+  for_parallel(i, numViews)
+  {
+    views[i].trans = (views[i].trans - pcCentre.pos) * normScale;
+  }
+}
+
+void Domset::deNormalizePointCloud()
+{
+  const size_t numPoints(points.size());
+// denormalizing points
+#if DOMSET_USE_OPENMP
+#pragma omp parallel for
+#endif
+  for_parallel(i, numPoints)
+  {
+    points[i].pos = (points[i].pos / normScale) + pcCentre.pos;
+  }
+
+  const size_t numOldPoints(origPoints.size());
+#if DOMSET_USE_OPENMP
+#pragma omp parallel for
+#endif
+  for_parallel(i, numOldPoints)
+  {
+    origPoints[i].pos = (origPoints[i].pos / normScale) + pcCentre.pos;
+  }
+
+  // denormalizing camera centers
+  const size_t numViews(views.size());
+#if DOMSET_USE_OPENMP
+#pragma omp parallel for
+#endif
+  for_parallel(i, numViews)
+  {
+    views[i].trans = (views[i].trans / normScale) + pcCentre.pos;
+  }
+}
+
+void Domset::voxelGridFilter(const float &sizeX, const float &sizeY, const float &sizeZ)
+{
+  if (sizeX <= 0.0f || sizeY <= 0.0f || sizeZ <= 0.0f)
+  {
+    std::cerr << "Invalid voxel grid dimensions error.\n";
+    exit(0);
+  }
+
+  Point minPt;
+  Point maxPt;
+  const size_t numP = points.size();
+  // finding the min and max values for the 3 dimensions
+  const float mi = std::numeric_limits<float>::min();
+  const float ma = std::numeric_limits<float>::max();
+  minPt.pos << ma, ma, ma;
+  maxPt.pos << mi, mi, mi;
+
+  for (size_t p = 0; p < numP; p++)
+  {
+    const Point newSP = points[p];
+    if (newSP.pos(0) < minPt.pos(0))
+      minPt.pos(0) = newSP.pos(0);
+    if (newSP.pos(1) < minPt.pos(1))
+      minPt.pos(1) = newSP.pos(1);
+    if (newSP.pos(2) < minPt.pos(2))
+      minPt.pos(2) = newSP.pos(2);
+    if (newSP.pos(0) > maxPt.pos(0))
+      maxPt.pos(0) = newSP.pos(0);
+    if (newSP.pos(1) > maxPt.pos(1))
+      maxPt.pos(1) = newSP.pos(1);
+    if (newSP.pos(2) > maxPt.pos(2))
+      maxPt.pos(2) = newSP.pos(2);
+  }
+
+  // finding the number of voxels reqired
+  size_t numVoxelX = static_cast<size_t>(ceil(maxPt.pos(0) - minPt.pos(0)) / sizeX);
+  size_t numVoxelY = static_cast<size_t>(ceil(maxPt.pos(1) - minPt.pos(1)) / sizeY);
+  size_t numVoxelZ = static_cast<size_t>(ceil(maxPt.pos(2) - minPt.pos(2)) / sizeZ);
+
+  /* std::cout << "Max = " << maxPt.pos.transpose() << std::endl;
```

Comment:
Maybe it could be better if we integrate the bunch of cerr/cout here into your LOG system...

Reply:
indeed, for that you just need to include Common/Common.h and use GET_LOG()

Reply:
thanks! fixed in 6e431611556137bcec6e3ecdb2f12c2a29b24fd8